File: Python/Readability problem/readability.py
# initialize the variables
letters = sentences = 0
# input the text
text = input("Text: ")
# words is equal to len of te text splited
words = len(text.split())
# loop each char in the text
for c in range(len(text)):
    if text[c].isalpha():
        letters = letters + 1
    if text[c] in ['!', '?', '.'] and text[c - 2] not in ['M', 'm'] and text[c - 3] not in ['M', 'm']:
        sentences = sentences + 1
# calculus
L = letters / words * 100
S = sentences / words * 100

index = 0.0588 * L - 0.296 * S - 15.8
# round the grade
grade = round(index)
# output
if grade < 1:
    print("Before Grade 1\n")
elif grade >= 1 and grade <= 16:
    print(f"Grade {grade}\n")
else:
    print("Grade 16+\n")

File: C/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // declare the sum of characters
    int sum = 0;

    // declare the sum of words
    int words = 0;

    // declare the sum of sentences
    int sentence = 0;

    // declare de grade by index
    int grade = 0;


    // get input the text to be analyzed
    string text = get_string("Text: ");

    // loop for checking each char of the text
    for (int i = 0; i < strlen(text); i++)
    {
        // checks if the char in the actual position is a letter, lower or upper
        if ((text[i] >= 'a' && text[i] <= 'z') || (text[i] >= 'A' && text[i] <= 'Z') /*|| text[i] == '-'*/)
        {
            sum += 1;
        }
        if (text[i + 1] == ' ' || text[i + 1] == '\0')
        {
            words += 1;
        }
        if ((text[i] == '.' && text[i - 2] != 'M' && text[i - 2] != 'm' && text[i - 3] != 'M' && text[i - 3] != 'm') || text[i] == '!'
            || text[i] == '?')
        {
            sentence += 1;
        }
    }


    float L = (float) sum / (float) words * 100;
    // printf("%f\n", L);
    float S = (float) sentence / (float) words * 100;
    // printf("%f\n", S);

    float index = 0.0588 * L - 0.296 * S - 15.8;

    /*printf("%i letter(s)\n", sum);
    printf("%i word(s)\n", words);
    printf("%i sentence(s)\n", sentence);
    printf("%f\n", index);*/

    grade = round(index);

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 1 && grade <= 16)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Grade 16+\n");
    }
}